osquery/tables/system/darwin/account_policy_data.mm

```diff
@@ -0,0 +1,135 @@
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/sql/sqlite_util.h"
+#include "osquery/tables/system/system_utils.h"
+
+#import <OpenDirectory/OpenDirectory.h>
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+void genAccountPolicyDataRow(const std::string& uid, Row& r) {
+  pt::ptree tree;
+
+  ODSession* s = [ODSession defaultSession];
+  NSError* err = nullptr;
+  ODNode* root = [ODNode nodeWithSession:s name:@"/Local/Default" error:&err];
+  if (err != nullptr) {
+    TLOG << "Error with OpenDirectory node: "
+         << std::string([[err localizedDescription] UTF8String]);
+    return;
+  }
+
+  ODQuery* q =
+      [ODQuery queryWithNode:root
+              forRecordTypes:kODRecordTypeUsers
+                   attribute:kODAttributeTypeUniqueID
+                   matchType:kODMatchEqualTo
+                 queryValues:[NSString stringWithFormat:@"%s", uid.c_str()]
+            returnAttributes:@"dsAttrTypeNative:accountPolicyData"
+              maximumResults:0
+                       error:&err];
+  if (err != nullptr) {
+    TLOG << "Error with OpenDirectory query: "
+         << std::string([[err localizedDescription] UTF8String]);
+    return;
+  }
+
+  // Obtain the results synchronously, not good for very large sets.
+  NSArray* od_results = [q resultsAllowingPartial:NO error:&err];
+  if (err != nullptr) {
+    TLOG << "Error with OpenDirectory results: "
+         << std::string([[err localizedDescription] UTF8String]);
+    return;
+  }
+
+  for (ODRecord* re in od_results) {
+    NSError* attrErr = nullptr;
+
+    NSArray* userPolicyDataValues =
+        [re valuesForAttribute:@"dsAttrTypeNative:accountPolicyData"
+                         error:&attrErr];
+
+    if (err != nullptr) {
+      TLOG << "Error with OpenDirectory attribute data: "
+           << std::string([[attrErr localizedDescription] UTF8String]);
+      return;
+    }
+
+    if (![userPolicyDataValues count]) {
+      return;
+    }
+
+    std::string userPlistString =
+        [[[NSString alloc] initWithData:userPolicyDataValues[0]
+                               encoding:NSUTF8StringEncoding] UTF8String];
+
+    if (userPlistString.empty()) {
+      return;
+    }
+
+    if (!osquery::parsePlistContent(userPlistString, tree).ok()) {
+      TLOG << "Error parsing Account Policy data plist";
+      return;
+    }
+  }
+
+  r["uid"] = BIGINT(uid);
+
+  if (auto creationTime = tree.get_optional<std::string>("creationTime")) {
```

Comment:
Instead of assigning in if statements, can we do this in one line using a default value?

Like this: https://github.com/facebook/osquery/blob/master/osquery/tables/applications/darwin/browser_plugins.cpp#L75

Or is there a reason that won't work?